go/pkg/cas/upload.go

```diff
@@ -57,6 +69,48 @@ type PathSpec struct {
 	Exclude *regexp.Regexp
 }
 
+func (ps *PathSpec) deepCopy() *PathSpec {
```

Comment:
no need to use https://golang.org/pkg/regexp/#Regexp.Copy for Exclude?

Reply:
good point. Renamed to `copy` and add a comment.

Reply:
no longer applicable